### ISSUE TITLE
Font(s) converted to `Helvetica` when using `QgsMapLayer.loadNamedStyle("/some/style/without_helvetica.qml")`

### ISSUE THREAD

**Issue by MattF-NSIDC** [qgis/QGIS#53012]
#### TLDR

* We use PyQGIS to create a `.qgs` project file containing many pre-configured and styled layers from many data sources.
* We create styles by hand in QGIS, and we choose the `Sans Serif` font for our labels.
* We export those styles to `.qml` in the GUI and we can confirm `fontFamily="Sans Serif"` appears in those [`.qml` files](https://github.com/nsidc/qgreenland/blob/5047c56a14659768b638f9c6cce8462c346931c0/qgreenland/ancillary/styles/transparent_labeled_shape.qml) (not `Helvetica`). We save those files in our code repository.
* When our code is run to create the project file, we create `QgsMapLayer` objects, and [add styles to those with `QgsMapLayer.loadNamedStyle()`](https://github.com/nsidc/qgreenland/blob/5047c56a14659768b638f9c6cce8462c346931c0/qgreenland/util/qgis/layer.py#L177). Immediately after we do that, we can see with `QgsMapLayer.exportNamedStyle()` that the layer style does not use `Sans Serif`, but instead uses `Helvetica` (this font is **not installed** on the system which created the styles, or the system which ran the code)
* Finally, at the end of our process, a `.qgs` project file is created, and it contains many instances of `fontFamily="Helvetica"`, and zero instances of `Sans` or `sans`.
* When a user opens this file on a system without `Helvetica`, they get 99+ warnings in the viewport about the missing font.

#### Background

On the QGreenland project, we create a QGIS project file using PyQGIS from a large set of interdisciplinary layers. We maintain a set of `.qml` style files for styling those layers, and we aim for those styles to display properly on all operating systems. We've been doing this for about 3 years now, starting with QGIS 3.16 when it was the LTR release, continuing to 3.28 LTR today.

We typically create those styles in QGIS and export on a Linux machine that doesn't have `Helvetica` installed. Historically, we've chosen the `Sans Serif` font for these styles. Here's an [example style file](https://github.com/nsidc/qgreenland/blob/5047c56a14659768b638f9c6cce8462c346931c0/qgreenland/ancillary/styles/transparent_labeled_shape.qml); note that `fontFamily="Sans Serif"`. Based on this, there seems to be no issue with creating the style and exporting it. Things don't get weird until we try and load this style into a layer object.

#### The problem

Unbeknownst to us, the whole time PyQGIS has been converting those `Sans Serif` font family choices into `Helvetica` (a non-free font that's not present on some operating systems) when we [apply](https://github.com/nsidc/qgreenland/blob/5047c56a14659768b638f9c6cce8462c346931c0/qgreenland/util/qgis/layer.py#L177) the `.qml` style to a layer object using `QgsMapLayer.loadNamedStyle()`. This happens when we run the code on a Linux system that does not have `Helvetica` installed. When one tries to open the final `.qgs` project file, they'll get warnings if their system doesn't have `Helvetica` installed. We didn't notice this because at first, these warnings were displayed only in the Layer Properties' "Labels" menu.

![image](https://user-images.githubusercontent.com/18539827/236253161-5e1c5319-96c2-4425-8f4a-7f74ffdeea8e.png)

However, with 3.18 or 3.22 (forget which), these warnings started surfacing in the viewport in large numbers and we started getting reports from users.

![image](https://user-images.githubusercontent.com/18539827/236254093-5167bb03-0cd8-4f2e-8622-82f8797526df.png)

We found that indeed, in our `.qgs` project file, the string `Helvetica` was showing up numerous times despite having no instances of `Helvetica` in our entire codebase. I investigated where in our processing pipeline this was being introduced, and found, using `QgsMapLayer.exportNamedStyle()`, that `Helvetica` is being introduced after calling `QgsMapLayer.loadNamedStyle("/path/to/the/sans_serif_style.qml")`.

#### Our workaround

I'm manually changing these `.qml` files to replace `fontFamily="Sans Serif"` with `fontFamily="Open Sans"`, and now QGIS is using the correct font, and I expect would download (in QGIS versions >=3.28) that font on other operating systems that don't have it.

### Steps to reproduce the issue

1. Create a `QgsMapLayer` called `map_layer` containing any polygon and give it a "label" attribute
2. Check its style *does not* contain `Helvetica` with `QgsMapLayer.exportNamedStyle()`, e.g.:

    ```
    from qgis.core import QgsReadWriteContext
    from PyQt5.QtXml import QDomDocument
    
    domdoc = QDomDocument()
    ctx = QgsReadWriteContext()
    map_layer.exportNamedStyle(domdoc, ctx)
    
    'Helvetica' in domdoc.toString()
    ```

3. Review [this style file](https://github.com/nsidc/qgreenland/blob/5047c56a14659768b638f9c6cce8462c346931c0/qgreenland/ancillary/styles/transparent_labeled_shape.qml) to confirm it does not contain any reference to `Helvetica`.
4. Apply the style from the previous step using `map_layer.loadNamedStyle("/path/to/style.qml")`
5. Repeat the check from step 2; but this time the layer's style *does* contain `Helvetica`.

If you add this layer to a project, the style from step 5 containing `Helvetica` will be present in the `.qgs` file.

### Versions

I've reproduced this issue using the steps above in QGIS >=3.16, <=3.28

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

Please let me know if I can provide any more information or help any other way :heart:

**Comment by nyalldawson** [comment]
There's nothing specifically in QGIS which is doing this replacement, and I can't reproduce this.

Try calling `QGuiApplication.font().family()` in the python console on a system where this is reproducible to see if for some reason the default font has been set to Helvetica.

**Comment by MattF-NSIDC** [comment]
Thanks for looking in to this! What operating system are you attempting to repro with? We're using a fairly out-of-the box Ubuntu 20 system without a desktop environment (the only font in `/usr/share/fonts` is "lato") to reproduce. I'm also reproducing consistently on my PopOS laptop.

```
>>> from PyQt5.QtGui import QGuiApplication
>>> qgui = QGuiApplication([])
>>> qgui.font().family()
```

returns `Sans Serif` on my PopOS laptop and `Noto Sans` on the Ubuntu 20 build machine (issue reproduces on both).

**Comment by MattF-NSIDC** [comment]
I believe I have provided the requested feedback. Can the `stale` and `feedback` labels be removed please @nyalldawson ? Thanks!

**Comment by alexbruy** [comment]
I also can't reproduce on Linux and Windows using latest master (6d58958a54a4ee4462052e23d5f260bd9454f27c) and LTR
